app/frontend/components/domains/requirement-template/builder-bottom-floating-buttons.tsx

```diff
@@ -1,22 +1,36 @@
 import { Button, Stack } from "@chakra-ui/react"
 import { ArrowUp } from "@phosphor-icons/react"
 import { t } from "i18next"
-import React from "react"
+import React, { useEffect, useState } from "react"
+import { handleScrollToTop } from "../../../utils/utility-functions"
 
 interface IProps {
-  onScrollToTop: () => void
-  onCollapseAll: () => void
+  onToggleCallback: (isCollapsedAll: boolean) => void
+  defaultAllCollapsed?: boolean
 }
 
-export function BuilderBottomFloatingButtons({ onScrollToTop, onCollapseAll }: IProps) {
+export function BuilderBottomFloatingButtons({ onToggleCallback, defaultAllCollapsed }: IProps) {
+  const [allCollapsed, setAllCollapsed] = useState(!!defaultAllCollapsed)
+  const [disabled, setDisabled] = useState(false)
+
+  useEffect(() => {
+    onToggleCallback(allCollapsed)
+    setTimeout(() => setDisabled(false), 500) // prevent double-clicking by accident while expanding/collapsing
+  }, [allCollapsed])
+
```

Comment:
@karenkun on a second read I don't even think we need to do this state syncing with the parent component. 
Remove internal state `const [allCollapsed, setAllCollapsed] = useState(!!defaultAllCollapsed)`. Just pass both the collapsed state and onClickDisable directly from parent, instead of trying to sync the states

Reply:
@taw27 I'm not sure I understand, the parent components dont seem to have any kind of collapse state. Are you sure this is right?